### PR TITLE
Review Task completed-Ready to Integrate

### PR DESCRIPTION
**Pull request by Jayasimha099**
Please review it as soon as possible before tonight. We will integrate the full application by tomorrow morning.

### CODE REVIEW

File: project1/templates/bookpage.html [hunk 1]
```diff
@@ -0,0 +1,118 @@
+<!DOCTYPE html>
+
+<html class="no-js">
+
+
+<head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
```
Comment (ashishk144) [inline project1/templates/bookpage.html:9]: Use a generic template and extend it to this file.

File: project1/application.py [hunk 2]
```diff
@@ -116,3 +120,47 @@ def userHome(Username):
...
+	else:
+		result = request.form.get("search")
+		result = '%'+result+'%'
+		search_result = Books.query.filter(or_(Books.tittle.ilike(result), Books.author.ilike(result), Books.isbn.ilike(result),Books.year.ilike(result))).all()
+		return render_template("search.html", books=search_result)
+
+
+@app.route("/bookpage/<username>/<isbn>", methods = ["POST","GET"])
+def bookpage(username, isbn):
+	user1 = username
```
Comment (ashishk144) [inline project1/application.py:139]: Check if the session varialble has a value or not, do not send the Username through the URI.

File: project1/application.py [hunk 3]
```diff
@@ -116,3 +120,47 @@ def userHome(Username):
...
+
+	if user1 in session:
+		user_reviews = reviewRate.query.filter_by(isbn = book_isbn).all()
```
Comment (ashishk144) [inline project1/application.py:142]: How is this query relevant when you're making the same queries in both if and else part again

File: project1/application.py [hunk 4]
```diff
@@ -116,3 +120,47 @@ def userHome(Username):
...
+				return render_template("bookpage.html",book=book,res=res,revie = user_reviews,username = user1)
+			return render_template("bookpage.html",book=book,message="You already given review!",revie = user_reviews,res=res,property="none",username=user1)
+
+		else:
+			rating = request.form.get("rating")
+			reviews = request.form.get("review")
+
+			isbn = book_isbn
+			username = user1
+
```
Comment (ashishk144) [inline project1/application.py:158]: How are you making sure that the user should be able to post exactly one review for one particular book?

**Review by ashishk144 — COMMENTED** [review]
Resolve all the conflicts

File: project1/templates/bookpage.html (hunk 1 above)
Comment (Jayasimha099) [reply project1/templates/bookpage.html:9]: added layout.html and remaining files are extended.

File: project1/application.py (hunk 2 above)
Comment (Jayasimha099) [reply project1/application.py:139]: I have already checked for username in session, but without username in URI it is not working we will implement that in integration part

File: project1/application.py (hunk 3 above)
Comment (Jayasimha099) [reply project1/application.py:142]: removed query in else part

File: project1/application.py (hunk 4 above)
Comment (Jayasimha099) [reply project1/application.py:158]: we are implementing that by checking if a review is given or not by if condition. And then if review given, we making the review block to hide using  "property- none". u can see it in lines 155

**Comment by Jayasimha099** [comment]
changes are made according to feedback. Ready for second review
